internal/proxy/proxy.go

```diff
@@ -510,19 +510,25 @@ func NewClient(ctx context.Context, d cloudsql.Dialer, l cloudsql.Logger, conf *
 	for _, inst := range conf.Instances {
 		m, err := c.newSocketMount(ctx, conf, pc, inst)
 		if err != nil {
-			for _, m := range mnts {
-				mErr := m.Close()
```

Comment:
If we return with an error anywhere below, we should still clean up any open sockets. So this code needs to be moved into a function and called in our two error exits below.

Reply:
Cleanup code has been added.